src/Database/LSMTree/Internal/IndexCompact.hs

```diff
@@ -385,8 +385,13 @@ newtype PageNo = PageNo { unPageNo :: Int }
   deriving stock (Show, Eq, Ord)
   deriving newtype NFData
 
--- TODO: Turn into newtype
-type NumPages = Int
+-- | The number of pages contained by an index or other paging data-structure.
+--
+-- Note: This is a 0-based number; take care to ensure arithmetic underflow
+-- does not occur during subtraction operations!
+newtype NumPages = NumPages Word
+  deriving stock (Eq, Ord, Show)
+  deriving newtype (Enum, Integral, NFData, Num, Real)
```

Comment:
I would argue that providing `Num`, `Real` and `Enum` instances removes a significant part of the advantages of having a newtype. They provide a lot of operations that don't make sense for this type, like `(*) :: NumPages -> NumPages -> NumPages` and can make conversions "too easy". It's safer to be more explicit, manually defining the small number of required operations, which are hopefully not too many. This of course might require more changes to the places where `NumPages` is used, but IMO it's often worth it.

Reply:
But generally I think it makes sense that you decided to work on this! It's a good idea to get more familiar with the code base by resolving TODOs and improving code quality.

Reply:
I concur -- let's remove the instances and wrap/unwrap the newtype where necessary

Reply:
I updated the PR by removing the permissive type-class instances and adds `getNumPages` as a polymorphic "unwrapper function."